appshell/appshell_extensions_mac.mm

```diff
@@ -391,6 +391,59 @@ void CloseLiveBrowser(CefRefPtr<CefBrowser> browser, CefRefPtr<CefProcessMessage
                                          );
 }
 
+int32 getSystemDefaultApp(ExtensionString fileTypes, ExtensionString& fileTypesWithdefaultApp)
+{
+
+    char delim[] = ",";
+    char separator[] = "##";
+    std::vector<ExtensionString> extArray;
+    char* token = std::strtok((char*)fileTypes.c_str(), delim);
+    while (token) {
+        extArray.push_back(token);
+        token = std::strtok(NULL, delim);
+    }
+
+    for (std::vector<ExtensionString>::iterator it = extArray.begin(); it != extArray.end(); ++it)
```

Comment:
Use a const iterator or
for (auto const & item : extArray)
{
}
or std::for_each.